sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConsumer.cs

```diff
@@ -267,15 +270,15 @@ internal class AmqpConsumer : TransportConsumer
 
                         // No events were available.
 
-                        return new List<EventData>(0);
+                        return EmptyEventSet;
```

Comment:
Personal preference is to use `Array.Empty<EventData>()` because it'll return the same instance rather than creating a varaible.. but it's because I'm lazy about F12s. :D

Reply:
Unless I'm misreading, I don't believe your statement is correct.  The source for [Array.Empty](https://github.com/microsoft/referencesource/blob/3b1eaf5203992df69de44c783a3eda37d3d4cd10/mscorlib/system/array.cs#L1086) indicates that it allocates on each [call](https://github.com/microsoft/referencesource/blob/3b1eaf5203992df69de44c783a3eda37d3d4cd10/mscorlib/system/array.cs#L3082).    Am I overlooking a cache somewhere?